src/Compiler/FSComp.txt

```diff
@@ -1509,7 +1509,7 @@ notAFunctionButMaybeDeclaration,"This value is not a function and cannot be appl
 3234,chkNoSpanLikeVariable,"The Span or IsByRefLike variable '%s' cannot be used at this point. This is to ensure the address of the local value does not escape its scope."
 3235,chkNoSpanLikeValueFromExpression,"A Span or IsByRefLike value returned from the expression cannot be used at ths point. This is to ensure the address of the local value does not escape its scope."
 3236,tastCantTakeAddressOfExpression,"Cannot take the address of the value returned from the expression. Assign the returned value to a let-bound value before taking the address."
-3237,tcCannotCallExtensionMethodInrefToByref,"Cannot call the byref extension method '%s. The first parameter requires the value to be mutable or a non-readonly byref type."
+3237,tcCannotCallExtensionMethodInrefToByref,"Cannot call the byref extension method '%s. 'this' parameter requires the value to be mutable or a non-readonly byref type."
```

Comment:
Not necessarily for this pr, but for future improvements - wondering if we can capture the actual name of the "instance" extension is being called on?

Reply:
Mhm. Yeah it's a pity that this first parameter doesn't have any special term/moniker.

Honestly I am not sure this change makes things clearer. I am just imaging the new wording in the tooltip above and I would be probably equally confused. IMO we should do what Vlad suggests, that will make a real difference.

Reply:
What if an extension method is called not on an identifier but on a some expression?

Reply:
> What if an extension method is called not on an identifier but on a some expression?

It can be detected, if it's an ident, then use its name and range, otherwise pass "this", but still expression range.